File: app/src/main/java/com/simbirsoft_mobile_revolution2017_v_team/mobilelibrary/services/BaseResponse.java
package com.simbirsoft_mobile_revolution2017_v_team.mobilelibrary.services;

import com.google.gson.annotations.SerializedName;

/**
 * Created by Olegka on 11.10.2017.
 */

public class BaseResponse<T> {

    @SerializedName("data")
    protected T data;

    public T getData() {
        return data;
    }

}

File: app/src/main/java/com/simbirsoft_mobile_revolution2017_v_team/mobilelibrary/services/BaseResponse.kt
package com.simbirsoft_mobile_revolution2017_v_team.mobilelibrary.services

import com.google.gson.annotations.SerializedName

/**
 * Created by user on 15.11.2017.
 */
class BaseResponseK<T> {

    @SerializedName("data")
    var data: T? = null
        private set

}